### DISCUSSION TITLE
Component definition should be in capital case

### DISCUSSION THREAD

**Discussion by gargajit** [fastn-stack/fastn#628] [Ideas & RFCs]
Component definition should be in capital case, such as 
```ftd
-- Component Hero:
```

Suggested by: @saurabh-lohiya

**Comment by amitu** [comment]
Problem is if we do this, it won't be just component names, but record names as well, file names etc. And as soon as you start using capital cases, you have to use full CamelCases. We can keep the routes lower cased. But in our case all files are valid routes. We expect you to code and documentation for component both in same file eg `components/button.ftd`, so if we made it `components/Button.ftd`, the URL would be `mysite.com/components/Button/`, which I am not sure I like. I mean what should we do if someone visits `mysite.com/components/button/`, give 404? Or make the URL fully case insensitive?

Not ruling it out, but its a bit too much dependent on taste, and I am basing my decisions on good looking URLs as well.
